src/controllers/UserNewsController.js

```diff
@@ -0,0 +1,155 @@
+const axios = require("axios");
+const userNewsPreferences = require("../data/userNewsPreferences.json");
+const userReadNews = require("../data/userReadNews.json");
+const userFavouriteNews = require("../data/userFavouriteNews.json");
+const { updateUserNewsPreferences } = require("../helpers");
+const Validation = require("../utilities/Validation");
+const redisClient = require("../cache/redis");
+const { newsURL } = require("../config");
+
+const parsedUserPreferences = JSON.parse(JSON.stringify(userNewsPreferences));
+
+const fetchNews = async (req, res) => {
+  const userId = req.id;
+  const userPreference = parsedUserPreferences[userId];
+
+  if (userPreference) {
+    await redisClient
```

Comment:
You can move this to a separate interface which will be responsible for talking to the Redis client only. You can call that file UserPreferenceCache. It will have getters and setters specifically for user preference functionality.